components/collector/src/source_collectors/harbor/security_warnings.py

```diff
@@ -48,23 +48,47 @@ async def _get_source_responses(self, *urls: URL) -> SourceResponses:
         projects_responses = await super()._get_source_responses(projects_api)
         for projects_response in projects_responses:
             for project in await projects_response.json():
+                if self._skip_project(project["name"]):
+                    continue
                 repositories_api = URL(f"{projects_api}/{project['name']}/repositories")
                 responses.extend(await self._get_source_responses_for_project(project, repositories_api))
         return responses
 
+    def _skip_project(self, project_name: str) -> bool:
+        """Return whether the project should be skipped."""
+        projects_to_ignore = self._parameter("projects_to_ignore")
+        if projects_to_ignore and match_string_or_regular_expression(project_name, projects_to_ignore):
+            return True
+        projects_to_include = self._parameter("projects_to_include")
+        if projects_to_include and not match_string_or_regular_expression(project_name, projects_to_include):
+            return True
+        return False
+
     async def _get_source_responses_for_project(self, project, repositories_api: URL) -> SourceResponses:
         """Return the source responses for a specific project."""
         responses = SourceResponses()
         repositories_responses = await super()._get_source_responses(repositories_api)
         for repositories_response in repositories_responses:
             for repository in await repositories_response.json():
                 repository_name = repository["name"].removeprefix(f"{project['name']}/")
+                if self._skip_repository(repository_name):
```

Comment:
The next line does some string munging, which is currently used to do the filtering.

Reply:
The next line quotes the repository name so it works when used in the URL for the artifacts endpoint. The filtering is done on the unquoted repository name.